packages/gatsby-source-drupal/src/normalize.js

```diff
@@ -12,11 +13,131 @@ const getHref = link => {
 
 exports.getHref = getHref
 
-const nodeFromData = (datum, createNodeId, entityReferenceRevisions = []) => {
+const imageCDNState = {
+  foundPlaceholderStyle: false,
+  hasLoggedNoPlaceholderStyle: false,
+}
+
+exports.imageCDNState = imageCDNState
+
+/**
+ * This FN takes in node data and returns Gatsby Image CDN fields that should be added to that node. If the input node isn't an image an empty object is returned.
+ */
+const getGatsbyImageCdnFields = async ({
+  node,
+  type,
+  pluginOptions,
+  fileNodesExtendedData,
+  reporter,
+}) => {
+  if (!pluginOptions.skipFileDownloads) {
+    return {}
```

Comment:
@smthomas I'm thinking massive sites might see some downsides to the `probeImageSize` calls if they're not skipping file downloads. So we can say image CDN only works if `skipFileDownloads` is enabled to mitigate that. curious what you think of this!

Reply:
I think this can work! They should either use Image CDN (which doesn't require local files) or not... so making it only work if they disable file downloads is probably a good idea.